octodns_powerdns/__init__.py

```diff
@@ -61,16 +61,6 @@ def __init__(
         *args,
         **kwargs,
     ):
-        super().__init__(id, *args, **kwargs)
-
-        if getattr(self, '_get_nameserver_record', False):
-            raise ProviderException(
-                '_get_nameserver_record no longer '
-                'supported; instead migrate to using a '
-                'dynamic source for zones; see '
-                'CHANGELOG.md'
-            )
-
```

Comment:
Why was this section moved?

Reply:
I have chosen to pass the new arguments as kwargs. For that reason I had to pop them out so these will not be passed to BaseProvider class which does not expect them. Because both soa_edit_api and mode_of_operation are instance attributes, when initializing these are calling powerdns_version method which needs all the attributes required by requests.

While the later is true, to keep the position of super() I can revert and simply define the two as positional arguments in the __init__